src/axios-har-tracker.ts

```diff
@@ -1,81 +1,77 @@
-const axios = require('axios').default;
-import { writeFileSync } from 'fs';
+import { AxiosStatic } from 'axios';
 import * as cookie from 'cookie';
 
-const date = new Date();
-const startDate = date.toISOString();
-
-let generatedHar = {
-  log: {
-    version: '1.2',
-    creator: {
-      name: 'axios-tracker',
-      version: '1.0.0'
-    },
-    pages: [],
-    entries: []
-  }
-};
-
-let newEntry = {
-  request: {},
-  response: {},
-  startedDateTime: startDate,
-  time: -1,
-  cache: {},
-  timings: {
-    blocked: -1,
-    dns: -1,
-    ssl: -1,
-    connect: -1,
-    send: 10,
-    wait: 10,
-    receive: 10,
-    _blocked_queueing: -1
-  }
-};
+export class AxiosHarTracker {
 
-async function axiosTracker(call) {
-  axios.interceptors.request.use(
-    async config => {
-      config.validateStatus = function () {
-        return true;
-      };
-      config.headers['request-startTime'] = process.hrtime();
-      const fullCookie = JSON.stringify(config.headers['Cookie']);
-      const version = config.httpVersion === undefined ? 'HTTP/1.1' : 'HTTP/' + config.httpVersion;
-
-      newEntry.request = {
-        method: config.method,
-        url: config.url,
-        httpVersion: version,
-        cookies: getCookies(fullCookie),
-        headers: [],
-        queryString: getParams(config.params),
-        headersSize: -1,
-        bodySize: -1
-      };
-      return config;
-    },
-    error => {
-      return Promise.reject(error);
-    }
-  );
+  private axios: AxiosStatic;
+  private generatedHar;
```

Comment:
Create interfaces for generatedHar and newEntry and use them as types for those properties